kimu/core/rectangular_tool.py

```diff
@@ -0,0 +1,403 @@
+import math
+from decimal import Decimal
+from typing import List
+
+from qgis.core import (
+    QgsFeature,
+    QgsField,
+    QgsGeometry,
+    QgsPointXY,
+    QgsProject,
+    QgsSnappingConfig,
+    QgsTolerance,
+    QgsVectorLayer,
+    QgsWkbTypes,
+)
+from qgis.gui import QgisInterface, QgsMapToolEmitPoint, QgsSnapIndicator
+from qgis.PyQt.QtCore import QVariant
+from qgis.PyQt.QtGui import QColor
+from qgis.utils import iface
+
+from ..qgis_plugin_tools.tools.custom_logging import setup_logger
+from ..qgis_plugin_tools.tools.i18n import tr
+from ..qgis_plugin_tools.tools.resources import plugin_name
+from ..ui.rectangular_dockwidget import RectangularDockWidget
+from .select_tool import SelectTool
+
+LOGGER = setup_logger(plugin_name())
+
+
+class RectangularMapping(SelectTool):
+    def __init__(
+        self, iface: QgisInterface, dock_widget: RectangularDockWidget
+    ) -> None:
+        super().__init__(iface)
+        self.ui: RectangularDockWidget = dock_widget
+        # Set suitable snapping settings
```

Comment:
Identical QgsSnappingConfig is already defined in the constructor of IntersectionLineCircle. Suggestion: move the definition to its own module so it can be easily used by both tools.